src/custom/components/AffiliateStatusCheck/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useActiveWeb3React } from 'hooks/web3'
+import NotificationBanner from 'components/NotificationBanner'
+import { useReferralAddress, useResetReferralAddress } from 'state/affiliate/hooks'
+import { updateAppDataHash } from 'state/affiliate/actions'
+import { useAppDispatch } from 'state/hooks'
+import { hasTrades } from 'utils/trade'
+import { uploadAppDataDoc } from 'utils/operator'
+import { generateReferralMetadataDoc } from 'utils/metadata'
+
+enum AffiliateStatus {
+  NOT_CONNECTED = 'NOT_CONNECTED',
+  OWN_LINK = 'OWN_LINK',
+  ALREADY_TRADED = 'ALREADY_TRADED',
+  ACTIVE = 'ACTIVE',
+}
+
+const STATUS_TO_MESSAGE_MAPPING: Record<AffiliateStatus, string> = {
+  NOT_CONNECTED: 'Please connect your wallet to participate',
+  OWN_LINK: 'You followed your own referral link',
+  ALREADY_TRADED: 'You have already traded with the current account',
+  ACTIVE: 'Your affiliate link will be effective on next trade',
+}
+
+export default function AffiliateStatusCheck() {
+  const appDispatch = useAppDispatch()
+  const resetReferralAddress = useResetReferralAddress()
+  const history = useHistory()
+  const { account, chainId } = useActiveWeb3React()
+  const referralAddress = useReferralAddress()
+  const [affiliateState, setAffiliateState] = useState<AffiliateStatus | undefined>()
+
+  useEffect(() => {
+    if (!referralAddress) {
+      return
+    }
+
+    if (referralAddress && !account) {
+      setAffiliateState(AffiliateStatus.NOT_CONNECTED)
+      return
+    }
+
+    if (referralAddress === account) {
+      // clean-up saved referral address if the user follows its own referral link
+      resetReferralAddress()
+      history.push('/profile')
+      setAffiliateState(AffiliateStatus.OWN_LINK)
+      return
+    }
+
+    async function validateUserHasTraded() {
```

Comment:
The function is clear in what it does, but the name does not reflect that.
It validates the trade AND uploads hash AND updates state etc

Not sure what would be a better name though.

Reply:
Yes, this happens when you create the initial function and name it and then you extend it without changing its name =p